python/GafferUI/SpreadsheetUI/_PlugTableView.py

```diff
@@ -450,17 +451,21 @@ def __keyPress( self, widget, event ) :
 
 	def __returnKeyPress( self ) :
 
-		selectionModel = self._qtWidget().selectionModel()
+		# If the selection is presented as a checkbox, toggle rather than
+		# opening the edit window.  This matches the single-click toggle mouse
+		# interaction.
 
-		# If we have a single index selected, and that is presented as
-		# a checkbox, toggle rather than opening the edit window.
-		# This matches the single-click toggle mouse interaction.
+		selectionModel = self._qtWidget().selectionModel()
 		selectedIndexes = selectionModel.selectedIndexes()
-		if len( selectedIndexes ) == 1 and selectionModel.model().presentsCheckstate( selectedIndexes[0] ) :
-			plug = selectionModel.model().valuePlugForIndex( selectedIndexes[0] )
-			if plug is not None :
-				with self.ancestor( GafferUI.PlugValueWidget ).getContext() :
-					_Algo.setPlugValues( [ plug ], not plug.getValue() )
+
+		if not selectedIndexes :
+			return
+
+		model = selectionModel.model()
+		if all( [ model.presentsCheckstate( i ) for i in selectedIndexes ] ) :
+			valuePlugs = [ model.valuePlugForIndex( i ) for i in selectedIndexes ]
+			with self.ancestor( GafferUI.PlugValueWidget ).getContext() :
```

Comment:
I don't think this context scope is doing anything is it?

Reply:
Removed [here](https://github.com/GafferHQ/gaffer/pull/4001/commits/34370b00ca3a845f3dea399885dd22f850261ee1).